src/dfx-core/src/canister/mod.rs

```diff
@@ -0,0 +1,20 @@
+use candid::Principal;
+use ic_agent::Agent;
+use ic_utils::interfaces::WalletCanister;
+
+use crate::error::canister::CanisterBuilderError;
+
+pub async fn build_wallet_canister(
+    id: Principal,
+    agent: &Agent,
+) -> Result<WalletCanister<'_>, CanisterBuilderError> {
+    Ok(WalletCanister::from_canister(
+        ic_utils::Canister::builder()
+            .with_agent(agent)
+            .with_canister_id(id)
+            .build()
+            .unwrap(),
```

Comment:
is this alright?

Reply:
fn_context made it very easy to add context at every level of the call stack.  I'm not sure we're adding much value here by adding an error type that mostly identifies the method being called.

What would you think about having this method's error type be `AgentError`?